codegen/templates/templates.go

```diff
@@ -167,7 +168,30 @@ func Render(cfg Options) error {
 		return err
 	}
 
-	cfg.Packages.Evict(code.ImportPathForDir(filepath.Dir(cfg.Filename)))
+	// Fieldalign the file's structs to save memory.
+	// TODO: if (cfg.fieldalignment enabled)
+	fileDir := filepath.Dir(cfg.Filename)
+
+	// change the current working directory to ensure the file is in the current module.
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	if err := os.Chdir(fileDir); err != nil {
+		return err
+	}
+
+	cmd := exec.Command("fieldalignment", "-fix", cfg.Filename)
```

Comment:
For cases like `fieldalignment` (or for example, `addlicense`), I add another line to `go generate` file and run it separately. I recommend not adding it to gqlgen, but if others think differently, it should be executed as a package (see [example](https://cs.opensource.google/go/x/tools/+/refs/tags/v0.1.10:go/analysis/passes/fieldalignment/cmd/fieldalignment/main.go)), because some users may not have it installed.

Reply:
Great! Didn't see the exported `Run: run` for some reason. I will change this if the idea is approved. Is there any reason you recommend not adding it as an option? I plan to add the boolean option here below `autobind`: https://gqlgen.com/config/

Reply:
Hey, @switchupcb, as a package maintainer of [ent](https://github.com/ent/ent), I can share my 2 cents why I didn't add it to the project, even not as an option. 

1\. For some packages, like gqlgen and ent, I never used `fieldalignment` because I prefer the order of the fields in the generated code to follow the same order they are defined in the graphql.schema or ent/schema. In most cases, I prefer code readability and maintainability over fine-tuning optimization. This is why it's not enabled by default.

2\. The reason it's not an option in ent, is because for users, adding the following line to the `go generate` process is simpler than all other options. Also, this gives them the option to choose any version they want (of `fieldalignment` for example) and may run it on multiple packages at once. 

```diff
//go:generate go run entgo.io/ent/cmd/ent generate ./ent/schema
//go:generate go run -mod=mod github.com/99designs/gqlgen
+//go:generate go run golang.org/x/tools/go/analysis/passes/fieldalignment/cmd/fieldalignment -fix <packages...>
```

Reply:
That seems more practical. You _can_ run it programmatically _(not using cmd)_, but its not straightforward.